firebase/test/core/data/repositories/stored-collections-repository.spec.ts

```diff
@@ -0,0 +1,128 @@
+import * as assert from "assert";
+import * as sinon from "sinon";
+import { SchemaValidator } from "#data/schemas/schema-validator";
+import SerializationError from "#faults/errors/serialization-error";
+import { StoredPublicCollection } from "#domain/models/collection";
+import { StoredCollectionsRepository } from "#data/repositories/stored-collections-repository";
+import { FirestoreGateway } from "#data/gateways/firestore-gateway";
+
+describe("StoredCollectionsRepository", () => {
+  let sandbox: sinon.SinonSandbox;
+  let firestoreGateway: sinon.SinonStubbedInstance<FirestoreGateway>;
+  let schemaValidator: sinon.SinonStubbedInstance<SchemaValidator>;
+  let storedCollectionsRepo: StoredCollectionsRepository;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    firestoreGateway = sandbox.createStubInstance(FirestoreGateway);
+    // Mocks the transaction function to always run what's is inside the context
+    firestoreGateway.runTransaction.callsFake(async (context) => {
+      await context();
+    });
+    schemaValidator = sandbox.createStubInstance(SchemaValidator);
+    storedCollectionsRepo = new StoredCollectionsRepository(
+      <FirestoreGateway>(<any>firestoreGateway),
+      <SchemaValidator>(<any>schemaValidator)
+    );
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe("setCollections", () => {
+    it("should update Collections set inside a single transaction", async () => {
+      const mockCollection = _newCollection();
+
+      await storedCollectionsRepo.setCollections([mockCollection]);
+
+      assert.ok(firestoreGateway.runTransaction.calledOnce);
+    });
+
+    it("should throw when one of the Collections has invalid format", () => {
+      const malformedCollection: StoredPublicCollection = <any>{ id: "id2", foo: "bar" };
+      const mockCollections = [_newCollection(), malformedCollection];
+
+      assert.rejects(() => storedCollectionsRepo.setCollections(mockCollections), SerializationError);
+    });
+
+    it("should update Collections using their raw representation", async () => {
```

Comment:
```suggestion
    it("should update a collection using their raw representation", async () => {
```

Following https://github.com/olmps/memo/pull/256/files#r762997476

Reply:
Updated to `should set collections using their raw representation` since it's not a single collection